### PR TITLE
ocamlPackages.piaf: init 0.1.0

### PR DESCRIPTION
**Pull request by superherointj**
* ocamlPackages.gluten: init 0.2.1
* ocamlPackages.gluten-lwt: init 0.2.1
* ocamlPackages.gluten-lwt-unix: init 0.2.1
* ocamlPackages.piaf: init 0.1.0

## Notes

`gluten-mirage` and `gluten-async` is not included because it is broken in this release. ~~On a next working release I may include it.~~

To test versions:
```
NIX_PATH=$HOME/git/ nix-shell -p  \
    ocaml-ng.ocamlPackages_4_13.findlib \
    ocaml-ng.ocamlPackages_4_13.gluten \
    ocaml-ng.ocamlPackages_4_13.gluten-lwt \
    ocaml-ng.ocamlPackages_4_13.gluten-lwt-unix \
    ocaml-ng.ocamlPackages_4_13.piaf \
    --run 'ocamlfind list' \
    | grep -e gluten -e gluten-lwt -e gluten-lwt-unix -e piaf
```

### CODE REVIEW

File: pkgs/development/ocaml-modules/gluten/default.nix [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{ callPackage
+, bigstringaf
+, faraday
+}:
+
+(callPackage ./generic.nix { }) {
```
[the remark this answers is not in the document]
Comment (vbgl) [reply pkgs/development/ocaml-modules/gluten/default.nix:6]: The problem is the complexity of the set-up: there might be unexpected behaviors (say, how to override the `lib` argument to `gluten` package? hopefully, nobody will need to do such a thing) or maintainability issues.

That being said, I don’t know if there is a better way to do… In similar cases, I tend to define the base package (here `gluten`) as usual and then make the derived package (e.g., `gluten-lwt`) `inherit` the parts that it shares with its “parent”. I’m not saying that you have to do it this way, I’m just looking for simple and efficient patterns.

[the remark this answers is not in the document]
Comment (superherointj) [reply pkgs/development/ocaml-modules/gluten/default.nix:6]: I will attempt to replace generic/callPackage with `inherit`. Please, wait.

[the remark this answers is not in the document]
Comment (superherointj) [reply pkgs/development/ocaml-modules/gluten/default.nix]: I have refactored the packages and added `piaf` which is the actual package which I was porting.

File: pkgs/development/ocaml-modules/gluten/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    faraday
+  ];
+
+  doCheck = false; # No tests
+
+  meta = {
+    description = "An implementation of a platform specific runtime code for driving network libraries based on state machines, such as http/af, h2 and websocketaf";
+    license = lib.licenses.bsd3;
+    homepage = "https://github.com/anmonteiro/gluten";
+    maintainers = with lib.maintainers; [ anmonteiro superherointj ];
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/gluten/default.nix:32]: Ping @anmonteiro.

Do you agree to be listed as maintainer here?

Comment (superherointj) [reply pkgs/development/ocaml-modules/gluten/default.nix:32]: I have an agreement with @anmonteiro that every time I package one of his packages I should add him as maintainer. I have been porting the packages from [his overlay](https://github.com/anmonteiro/nix-overlays).

Comment (anmonteiro) [reply pkgs/development/ocaml-modules/gluten/default.nix:32]: Confirmed

**Pull request merged**